torch/csrc/autograd/python_hook.cpp

```diff
@@ -169,15 +169,12 @@ static void check_single_result(PyObject* _original, PyObject* _result, PyObject
     throw std::runtime_error(ss.str());
   }
 
-  if (original.is_cuda() != result.is_cuda()) {
+  if (original.type().backend() != result.type().backend()) {
```

Comment:
I don't really understand what this test is doing, because I think the `type().ID()` test above would catch this.  But in any case, if you want the equivalent behavior this should be written in terms of devices, not backends.

Reply:
I have tried to minimize the changes in autograd, since I am not too familiar with it.
I had to change it since otherwise it'd fail when used with XLA devices/backends.

Reply:
is there something I can run to see the failure?  My take is the existing code is bogus and thus your code shouldn't matter, but I must be misunderstanding something if you can trigger it.

Reply:
Let me try again on this side, with the new code. Might be stray stuff, before I set up correctly the device guard.

Reply:
Did not trigger with proper device guard setup.
Removed this change with the new commit.